test/libyul/yulOptimizerTests/fullSuite/name_dependent_cse_bug_part_1.yul

```diff
@@ -35,10 +35,9 @@
 //         {
 //             let _1 := add(a, a)
 //             let var := add(_1, _1)
-//             let mid := var
 //             switch a
 //             case 0 { a := var }
-//             default { sstore(0, mid) }
+//             default { sstore(0, var) }
```

Comment:
This is what the fix does.

Reply:
Nice :D

However, the test suite seems to be adding an extra `let a_1 := 0`, I suspected that it is the use of `fullSuite` there that may have a different optimization steps than the one used by `--optimize` flag?. Compiling the example by hand yields the exact test case you added.

```
object "object" {
    code {
        {
            let a := 0
            for { } a { }
            {
                let _1 := add(a, a)
                let var := add(_1, _1)
                switch a
                case 0 { a := var }
                default { sstore(0, var) }
            }
        }
    }
}
```

Reply:
It seems to be a difference between EVM versions. It gets added on pre-`shanghai` EVMs, while my version is from `shanghai`. Not entirely sure why exactly but some steps do have output dependent on EVM version so this does not seem to be a bug.


I split this test into two versions to account for this and now CI should pass.